tests/sync.rs

```diff
@@ -203,6 +203,26 @@ fn try_recv() {
     })
 }
 
+#[cfg(feature = "std")]
+#[test]
+fn try_recv_then_drop_receiver() {
+    maybe_loom_model(|| {
+        let (sender, receiver) = oneshot::channel::<u128>();
+        let t1 = thread::spawn(move || {
+            let _ = sender.send(42);
+        });
+        let t2 = thread::spawn(move || {
+            assert!(matches!(
+                receiver.try_recv(),
+                Ok(42) | Err(TryRecvError::Empty)
+            ));
+            mem::drop(receiver);
+        });
+        t1.join().unwrap();
+        t2.join().unwrap();
```

Comment:
Is this supposed to test the race condition that the library had before this PR? I can't make this test fail even when I backport it to current `master`. Of course a normal run won't trigger it very often at all. But if it really was a thread interleaving issue loom should pick it up, no?

Would be nice to understand the situation and have a test that actually catch the issue. Otherwise it won't really protect against regressions.

Reply:
I think this was to test the interleaving of recv fails, send happens, receiver is dropped. I think I added this test while I was working out how memory reclamation worked in this crate since I didn't see a test that tested that ordering.